### DISCUSSION TITLE
Symfony UX - Update PostRender

### DISCUSSION THREAD

**Discussion by smbpunt** [symfony/symfony#49699] [Q&A]
Hi, 

I’m having trouble with Symfony UX and re-rendering the component PostRender.
What I’d like to do is to load the component with no data and a text “Loading...“, call right-after first render a loadData() function which call an API, and then render the result.

I’ve succeeded doing this with a button:

```
#[AsLiveComponent('topic_list')]
class TopicListComponent
{
    use DefaultActionTrait;
    private array $topics = [];
    private bool $isLoaded = false;

    public function __construct(private readonly MyApiClient $client) {}

    #[LiveAction]
    public function loadData(): void
    {
        $this->topics = $this->client->getTopicList(useCache: false);
        $this->isLoaded = true;
    }

    // getters...
```

Twig : 

```
<div {{ attributes }}>
    <button data-action="live#action" data-action-name="loadData">Load</button>
    {% if this.isLoaded %}
        <ul>
            {% for topic in this.topics %}
                <li>
                    {{ topic.label }}
                </li>
            {% endfor %}
        </ul>
    {% else %}
        <p>Loading</p>
    {% endif %}
</div>
```

Now I'd like to remove this button.

I've tried with EventSubscriberInterface and PostRender or PostMountEvent and nothing is happening, I still have the loading text even tho loadData() is called...

Any suggestion ?

Thanks

**Comment by DZunke** [comment]
I have searched for the same idea but then found out there is no real async post render loading supported yet. Played a bit around with a custom stimulus controller but at the end i found an issue within the ux project.

https://github.com/symfony/ux/issues/102

In this issue, as future G, there were the Turbo Frames mentioned. After a look to them i found out they are doing exactly what i wanted with just a controller action and a frame within the frontend. I then loaded the content with a Twig (non live) Component in the replacement frame and it worked like a charme.
